### PR TITLE
sbt script: Favor java's `user.home` over `$HOME`

### PR DESCRIPTION
**Pull request by rdesgroppes**
JVM powered applications may be given an alternate home by means of the
`user.home` system property, which is handy for managing caches in CI
(the property may be set by different means, including environment
variables such as `JAVA_TOOL_OPTIONS` or `_JAVA_OPTIONS`).

Alas, this doesn't fully work when the `sbt` script downloads the
launcher jar to `$HOME`.

The present change consists in retrieving the value of this property by
means of a `findProperty` function extracted from the existing
`getPreloaded` one (adapted accordingly).

No java process get spawned here.

### CODE REVIEW

File: sbt [hunk 1]
```diff
@@ -117,6 +117,16 @@ download_url () {
   } && [[ -f "$jar" ]]
 }
 
+user_home () {
+  local user_home
+  user_home=$("$java_cmd" -XshowSettings:properties -version 2>&1 | sed -n 's/\s*user.home = //p' 2>/dev/null)
```
Comment (ritschwumm) [inline sbt]: will this be called on every startup, or only when sbt is not installed yet?

Comment (rdesgroppes) [reply sbt]: It's executed from within a shell function, so: only when there's a need for resolving a path under user's home directory.

There are currently 2 cases: either to form a path where to download the launcher jar (if needed), or to sync "preloaded" jars from sbt's installation directory to user home directory (if needed).

Comment (eed3si9n) [reply sbt]: Each call to `time java -XshowSettings:properties -version` adds around 100ms.

For `--client` mode that could potentially double our triple the startup - https://github.com/sbt/sbt-launcher-package/pull/334

Comment (vegerot) [reply sbt]: Do we measure this anywhere?  It would have been nice if https://github.com/sbt/sbt-launcher-package/pull/334 also added tests for this

Comment (eed3si9n) [reply sbt]: No currently we don't have any benchmark or measurements on `sbt` script startup time.

Comment (rdesgroppes) [reply sbt]: OK, following above concern, I've amended the commit with a second approach that factors out property search.

**Review by eed3si9n — APPROVED** [review]
Thanks @rdesgroppes!

File: sbt [hunk 2]
```diff
@@ -301,39 +302,49 @@ jdk_version() {
...
+getPreloaded() {
+  local preloaded && preloaded=$(findProperty sbt.preloaded)
+  [ "$preloaded" ] && echo "$preloaded" && return
+
+  local global_base && global_base=$(findProperty sbt.global.base)
+  [ "$global_base" ] && echo "$global_base/preloaded" && return
 
-  echo "${via_explicit:-${via_global_base}}"
+  local user_home && user_home=$(findProperty user.home)
+  echo "${user_home:-$HOME}/.sbt/preloaded}}"
```
Comment (rdesgroppes) [inline sbt:347]: :warning: trailing `}}`, my bad!

Comment (rdesgroppes) [reply sbt:347]: => #6494.

**Pull request merged**